File: sample_app_dagger/src/main/java/com/chumarin/stanislav/sample_app_dagger/SplashActivity.java
package com.chumarin.stanislav.sample_app_dagger;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.chumarin.stanislav.sample_app_dagger.login.LoginActivity;
import dagger.android.AndroidInjection;
import vit.khudenko.android.sessiontracker.SessionRecord;
import vit.khudenko.android.sessiontracker.SessionTracker;

import javax.inject.Inject;
import java.util.List;
import java.util.stream.Collectors;

public class SplashActivity extends AppCompatActivity {

    @Inject
    SessionTracker<Session.Event, Session.State> sessionTracker;

    private Handler handler;

    private final Runnable action = () -> {
        List<SessionRecord<Session.State>> activeSessionRecords = sessionTracker.getSessionRecords().stream()
                .filter(stateSessionRecord -> stateSessionRecord.getState() == Session.State.ACTIVE)
                .collect(Collectors.toList());

        String currentSessionId = activeSessionRecords.isEmpty() ? null : activeSessionRecords.get(0).sessionId();

        if (currentSessionId == null) {
            startActivity(new Intent(this, LoginActivity.class));
        } else {
            Intent intent = MainActivity.createIntent(this, currentSessionId);
            startActivity(intent);
        }
        finish();
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        handler = new Handler(getApplication().getMainLooper());
    }

    @Override
    protected void onResume() {
        super.onResume();
        handler.postDelayed(action, 500);
    }

    @Override
    protected void onPause() {
        super.onPause();
        handler.removeCallbacks(action);
    }
}

File: sample_app_koin/src/main/kotlin/vit/khudenko/android/sessiontracker/sample/koin/SplashActivity.kt
package vit.khudenko.android.sessiontracker.sample.koin

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import org.koin.android.ext.android.inject
import vit.khudenko.android.sessiontracker.SessionTracker
import vit.khudenko.android.sessiontracker.sample.koin.login.LoginActivity

class SplashActivity : AppCompatActivity() {

    private val sessionTracker: SessionTracker<Session.Event, Session.State> by inject()

    private lateinit var handler: Handler

    private val action = Runnable {
        val activeSessionRecords = sessionTracker.getSessionRecords()
            .filter { record ->
                record.state == Session.State.ACTIVE
            }
        val currentSessionId = if (activeSessionRecords.isEmpty()) {
            null
        } else {
            activeSessionRecords.first().sessionId
        }
        if (currentSessionId == null) {
            startActivity(Intent(this, LoginActivity::class.java))
        } else {
            val intent = Intent(this, MainActivity::class.java)
            intent.putExtra(EXTRA_CURRENT_SESSION_ID, currentSessionId)
            startActivity(intent)
        }
        finish()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        handler = Handler(application.mainLooper)
    }

    override fun onResume() {
        super.onResume()
        handler.postDelayed(action, 500)
    }

    override fun onPause() {
        super.onPause()
        handler.removeCallbacks(action)
    }
}
